File: apps/web/src/confidence.ts
// The confidence contract (US-13). These thresholds mirror the backend's
// apps/api/app/confidence.py exactly — one contract, two renderers.
export const GREEN_MIN = 0.7;
export const YELLOW_MIN = 0.4;

export type Band = "green" | "yellow" | "red";

export function bandFromConfidence(confidence: number): Band {
  if (confidence >= GREEN_MIN) return "green";
  if (confidence >= YELLOW_MIN) return "yellow";
  return "red";
}

File: apps/api/app/confidence.py
"""The confidence contract (US-13): every surfaceable item exposes
{confidence, band, reason}. One threshold definition — the frontend chip
mirrors these exact numbers, and the triage scorer uses the same bands."""

GREEN_MIN = 0.7
YELLOW_MIN = 0.4


def band_for(confidence: float) -> str:
    if confidence >= GREEN_MIN:
        return "green"
    if confidence >= YELLOW_MIN:
        return "yellow"
    return "red"
